### PR TITLE
eclipses.plugins: Add Apache ivyde, iyvderv, ivyant and ivy plugins

### PR DESCRIPTION
**Pull request by r3dl3g**
Add plugins to use ivy in eclipse. To use it, dependent plugin zest has also to be added

###### Motivation for this change

I need ivy plugins in eclipse.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/eclipse/plugins.nix [hunk 1]
```diff
@@ -639,4 +639,92 @@ rec {
     };
   };
 
+  ivyde = buildEclipsePlugin rec {
+    name = "ivyde-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/eclipse/plugins.nix:643]: Does buildEclipsePlugin not accept pname + version?

File: pkgs/applications/editors/eclipse/plugins.nix [hunk 2]
```diff
@@ -639,4 +639,92 @@ rec {
...
+    srcPlugin = fetchurl {
+      url = "https://downloads.apache.org/ant/ivyde/updatesite/ivyde-${version}/plugins/org.apache.ivyde.eclipse_${version}.jar";
+      sha1 = "0c80c2e228a07f18efab1c56ea026448eda70c06";
+    };
+
+    meta = with lib; {
+      homepage = "https://ant.apache.org/ivy/ivyde/index.html";
+      description = "A plugin which integrates Apache Ivy's dependency management";
+      license = licenses.asl20;
+      platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/eclipse/plugins.nix:660]: meta should always contain a maintainer.

File: pkgs/applications/editors/eclipse/plugins.nix [hunk 3]
```diff
@@ -639,4 +639,92 @@ rec {
...
+    };
+
+    srcPlugin = fetchurl {
+      url = "https://downloads.apache.org/ant/ivyde/updatesite/ivyde-${version}/plugins/org.apache.ivyde.eclipse.resolvevisualizer_${version}.jar";
+      sha1 = "225e0c8ccb010d622c159560638578c2fc51a67e";
+    };
+
+    meta = with lib; {
+      homepage = "https://ant.apache.org/ivy/ivyde/index.html";
+      description = "A graph viewer of the resolved dependencies. It is an optional dependency of Apache IvyDE since it requires additional plugins to be installed (Zest).";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/eclipse/plugins.nix:680]: That is to long for description. Maybe convert it into longDescription and add a shorter description?

File: pkgs/applications/editors/eclipse/plugins.nix [hunk 4]
```diff
@@ -639,4 +639,92 @@ rec {
...
+    srcPlugin = fetchurl {
+      url = "https://downloads.apache.org/ant/ivyde/updatesite/ivy-${version}/plugins/org.apache.ivy.eclipse.ant_${version}.jar";
+      sha256 = "9e8ea20480cf73d0f0f3fb032d263c7536b24fd2eef71beb7d62af4e065f9ab5";
+    };
+
+    meta = with lib; {
+      homepage = "https://ant.apache.org/ivy/index.html";
+      description = "A popular dependency manager focusing on flexibility and simplicity";
+      license = licenses.asl20;
+      platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/eclipse/plugins.nix:704]: Shouldn't this be the platforms of eclipse? buildEclipsePlugin should default that to the platforms supported by eclipse.

File: pkgs/applications/editors/eclipse/plugins.nix [hunk 5]
```diff
@@ -639,4 +639,92 @@ rec {
...
+
+    meta = with lib; {
+      homepage = "https://ant.apache.org/ivy/ivyde/index.html";
+      description = "Ant Tasks integrated into Eclipse's Ant runtime";
+      license = licenses.asl20;
+      platforms = platforms.all;
+    };
+  };
+
 }
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/eclipse/plugins.nix:730]:
```suggestion
  };
}
```

File: pkgs/applications/editors/eclipse/plugins.nix (hunk 1 above)
Comment (r3dl3g) [reply pkgs/applications/editors/eclipse/plugins.nix:643]: This was my very first pull request for nixos and I simply copied from existing plugins. All existing plugins use that way for the name, so I believe it's the only way to do so.

File: pkgs/applications/editors/eclipse/plugins.nix (hunk 2 above)
Comment (r3dl3g) [reply pkgs/applications/editors/eclipse/plugins.nix:660]: As mentioned before, it was my first pull. So I didn't yet feel suitable as maintainer. But now I think I can add myself.
Btw. many other plugins also haven't a maintainer and some of them didn't work anymore. Should they be removed?

File: pkgs/applications/editors/eclipse/plugins.nix (hunk 4 above)
Comment (r3dl3g) [reply pkgs/applications/editors/eclipse/plugins.nix:704]: Once again I've just choose the same value, as all other plugins have. I have no idea what to choose else.

File: pkgs/applications/editors/eclipse/plugins.nix (hunk 5 above)
Comment (r3dl3g) [reply pkgs/applications/editors/eclipse/plugins.nix:730]: some add empty lines, some remove them... :-)
I'm fine to remove them.

File: pkgs/applications/editors/eclipse/plugins.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/editors/eclipse/plugins.nix:660]: > Btw. many other plugins also haven't a maintainer and some of them didn't work anymore. Should they be removed?

If there is an update that makes them work again they should be updated but if upstream no longer develops them and there are no fixes we can drop them.

File: pkgs/applications/editors/eclipse/plugins.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/editors/eclipse/plugins.nix:710]: Fine for now I guess.

**Pull request merged**